src/internal_events/common.rs

```diff
@@ -53,6 +53,30 @@ impl<'a> InternalEvent for EndpointBytesSent<'a> {
     }
 }
 
+#[derive(Debug)]
+pub struct SocketConnectionError<E> {
+    pub error: E,
+}
+
+impl<E: std::error::Error> InternalEvent for SocketConnectionError<E> {
+    fn emit(self) {
+        error!(
+            message = "Unable to connect.",
+            error = %self.error,
+            error_code = "failed_connecting",
+            error_type = error_type::CONNECTION_FAILED,
+            stage = error_stage::PROCESSING,
```

Comment:
Yeah, I agree on that. Even though these are currently only used in a sink context, I'll add `Outgoing` to the struct names so that the direction is clear

Reply:
👍 we had done something similar here: https://github.com/vectordotdev/vector/blob/master/src/internal_events/socket.rs#L104

Reply:
Done with eaa93e3086b01b062052f140ecea1b927b2518dc